### PR TITLE
Starts beluga implementation from scratch

### PR DESCRIPTION
**Pull request by silvestrpredko**
# This PR is for [RumMqtt](https://github.com/bytebeamio/rumqtt) integration.

@scootermon don't hesitate to write here your ideas and improvements.

### CODE REVIEW

File: beluga/src/mqtt/mod.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+#[derive(Debug, Clone)]
+pub struct MqttClient {
+    client: AsyncClient,
+    worker: Arc<Mutex<JoinHandle<()>>>,
+    subscribers: Arc<Mutex<HashMap<String, Sender<Publish>>>>,
+}
+
+impl MqttClient {
+    pub async fn subscribe(&self, topic: &str, qos: QoS) -> Result<Receiver<Publish>> {
```
Comment (scootermon) [inline beluga/src/mqtt/mod.rs:99]: We should make it very clear that the current implementation doesn't support wildcard topics.

**Review by scootermon — COMMENTED** [review]
It's very hard to review this PR because of the overwhelmhing amount of changes. Let's get a first version of this PR merged as soon as possible so the following iterations become a bit cleaner :)

**Comment by silvestrpredko** [comment]
> It's very hard to review this PR because of the overwhelmhing amount of changes. Let's get a first version of this PR merged as soon as possible so the following iterations become a bit cleaner :) 

Agree

**Comment by silvestrpredko** [comment]
@scootermon Maybe let's review and merge the current version and then improve the client in the next versions.

**Review by scootermon — APPROVED** [review]
Yes, let's just get this merged :D

**Pull request merged**